### ISSUE TITLE
[Suggestion] Option to reroll world in SP

### ISSUE THREAD

**Issue by ssjkakaroto** [diasurgical/devilutionX#74]
Hi, in Diablo multiplayer the world is always rerolled when you enter a game and in single-player it's never rerolled.
In Torchlight 2 the save game is the same for both, but instead of doing like Diablo 1, they added an option for [rerolling the world](https://www.urgametips.com/2012/11/torchlight-ii-how-to-replay-completed.html), which I think could useful for DevilutionX too.

Maybe it's worth considering unifying the save files like in TL2, specially because there is the option of playing "multiplayer" with loopback interface.

**Comment by CubeTheThird** [comment]
Are you looking to have an "automatic" reroll whenever you load up your singleplayer game? This sounds pretty much like the "new game" menu option, with the exception that a) your dropped items are preserved, and b) you have access to the same shortcuts to the dungeon. Correct?

**Comment by ssjkakaroto** [comment]
Yes, exactly. It makes much more sense with unified saves. I have edited the text accordingly.

**Comment by AJenbo** [comment]
Since you can already re-roll the would in SP I changed the issue to be about retaining the world in MP as that seams to be what is being requested here.

**Comment by seedee** [comment]
@diasurgical any progress on this

**Comment by AJenbo** [comment]
Doesn't look like anyone has tried it yet. My expectation is that it will work just fine if hooked up.

**Comment by MethodicalMadness** [comment]
Would be amazing to save mp worlds, anyone attempted this?

**Comment by qndel** [comment]
> Would be amazing to save mp worlds, anyone attempted this?

I don't see any point, there's singleplayer for that

**Comment by julealgon** [comment]
> I don't see any point, there's singleplayer for that

One of the points is to stop playing with a friend and resume later from where you stopped. It's fairly reasonable IMHO.

**Comment by MethodicalMadness** [comment]
> > Would be amazing to save mp worlds, anyone attempted this?
> 
> I don't see any point, there's singleplayer for that

They had coop in the playstation one version of the game. It was great. But I imagine that save was basically single player save system just saving an extra character. When you save in MP I imagine it is just saving the character like the save character option. Maybe if the host could just save quest progress and map seeds etc and anyone joining inherited the quest progress...

**Comment by AJenbo** [comment]
That's the idea. Basically treat it like there was a disconnect from the host and everyone rejoined.

**Comment by CleitonDeLima** [comment]
I would like to see this, it would be nice to continue the game with my friends.

**Comment by julealgon** [comment]
> I would like to see this

I'd recommend giving a thumbs up on the issue to signal your support @CleitonDeLima .

> Is there any chance this exists?

I'd say it's very likely it will be implemented at some point, though you can probably tell the group of devs working on DevilutionX is pretty small so this will compete with other features and may take a while to be fleshed out.
